add deploy-3 docs (#2633)
wso2/docs-is: en/docs/deploy/configure-a-read-write-active-directory-user-store.md

<td>Connection URL</td>
<td><p>This is the connection URL to the userstore server. In the case of default LDAP in Carbon, the port is specified in the <code><&ZeroWidthSpace;IS_HOME>/repository/conf/carbon.xml</code> file, and a reference to that port is included in this configuration.</p>
<p>Sample values:<br />
<a href="ldap://10.100.1.100:389">ldap://10.100.1.100:389</a><br />
<a href="ldaps://10.100.1.102:639">ldaps://10.100.1.102:639</a><br />
<br />
If you are connecting over ldaps (secured LDAP),<br />
you need to import the certificate of userstore to <code><&ZeroWidthSpace;IS_HOME>/repository/resources/security/client-truststore.jks</code>. For information on how to add certificates to the truststore and how keystores are configured and used in a system, see<br />
<a href="../../../deploy/security/use-asymmetric-encryption">Use asymmetric encryption.</a><br />
<br />
If LDAP connection pooling is used, see <br />
<a href="../../../deploy/performance/performance-tuning-recommendations#performance-tuning-ldaps-pooling">performance tuning ldaps pooling.</a></p></td>
</tr>
<tr class="odd">
<td>ConnectionName</td>
<td>connection_name</td>
<td>Connection Name</td>

> <td><p>This is the username that is used to connect to the userstore and perform various operations. This user does not need to be an administrator in the userstore or have an administrator role in WSO2 IS. However, this user MUST have permissions to read the user list and users' attributes and to perform search operations on the userstore. The value you specify is used as the DN (Distinguish Name) attribute of the user who has sufficient permissions to perform operations on users and roles in LDAP.</p>

Reviewer: ```suggestion
<td><p>This is the username that is used to connect to the userstore and perform various operations. This user does not need to be an administrator in the userstore or have an administrator role in WSO2 IS. However, this user must have permissions to read the user list and users' attributes and to perform search operations on the userstore. The value you specify is used as the DN (Distinguish Name) attribute of the user who has sufficient permissions to perform operations on users and roles in LDAP.</p>
```

<td>user_entry_object_class</td>
<td>User Entry Object Class</td>
<td>Object class used to construct user entries.<br />

> Default: identityPerson(a custom object class defined in WSO2 Identity Server)</td>

Reviewer: ```suggestion
Default: identityPerson (a custom object class defined in WSO2 Identity Server)</td>
```

</tr>
<tr class="odd">
<td>UserNameAttribute</td>
<td>user_name_attribute</td>
<td>Username Attribute</td>
<td><p>This is a uniquely identifying attribute that represents the username of the user. Users can be authenticated using their email address, UID, etc. The value of the attribute is considered as the username.</p>
<p>Default: uid<br />
<br />

> Note: email address is considered as a special case in WSO2 products, if you want to set the email address as username, see <a href="../../../guides/identity-lifecycles/enable-email-as-username">Enable using email address as the username.</a></p>

Reviewer: ```suggestion
Note: email address is considered as a special case in WSO2 Identity Server, if you want to set the email address as username, see <a href="../../../guides/identity-lifecycles/enable-email-as-username">Enable using email address as the username.</a></p>
```

PLAIN_TEXT - Plain text passwords.(Default)</p>
<p>If you just configure as SHA, It is considered as SHA-1, It is always better to configure algorithm with higher bit value as digest bit size would be increased.<br />
<br />

> Most of the LDAP servers (such as OpenLdap, OpenDJ, AD, ApacheDS and etc..) are supported to store password as salted hashed values (SSHA).<br />

Reviewer: ```suggestion
Most of the LDAP servers (such as OpenLdap, OpenDJ, AD, ApacheDS and etc.) are supported to store passwords as salted hashed values (SSHA).<br />
```

> Therefore, the WSO2IS server requires the password fed in to the connected userstore as a plain text value. Then the LDAP userstore can store them as salted hashed value. To feed the plain text into the LDAP server, you need to set PasswordHashMethod to “PLAIN_TEXT”.<br />

Reviewer: ```suggestion
Therefore, the WSO2 IS requires the password fed in to the connected userstore as a plain text value. Then the LDAP userstore can store them as a salted hashed value. To feed the plain text into the LDAP server, you need to set PasswordHashMethod to “PLAIN_TEXT”.<br />
```

> However, if your LDAP does not support to store user password as hashed values you can configure the WSO2 server to hash the password and feed the hashed password in to the LDAP server. To do this, you need to configure the PasswordHashMethod property with SHA (SHA-1), SHA-256, SHA-512. WSO2 server cannot create a salted hashed password (SSHA) to feed in to the LDAP.</p></td>

Reviewer: ```suggestion
However, if your LDAP does not support storing user passwords as hashed values you can configure the WSO2 server to hash the password and feed the hashed password into the LDAP server. To do this, you need to configure the PasswordHashMethod property with SHA (SHA-1), SHA-256, SHA-512. WSO2 server cannot create a salted hashed password (SSHA) to feed into the LDAP.</p></td>
```
